### PR TITLE
Enforce updating R documents

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Enforce updating R documents when R files are changed. This allows us to avoid rebuilding the R documents when we release a new version of MLflow and saves us about **10~15** min (installing R packages without cache takes a long time). After merging this PR, we need to update our internal Jenkins job.

## How is this patch tested?

- Existing tests
- Verify that the `build_doc_r` job fails when `source/R-api.rst` requires updates.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/document.R [hunk 1]
```diff
@@ -48,6 +48,8 @@ You can use the R API to `install MLflow <install_mlflow_>`_, start the `user in
     :depth: 1
 "
 rst_doc <- readLines("../../../docs/source/R-api.rst")
+# Convert non-breaking spaces inserted by pandoc to regular spaces
+rst_doc <- gsub("\302\240", " ", rst_doc)
```
Comment (harupy) [inline mlflow/R/mlflow/document.R:52]: This allows us to update `R-api.rst` by simply copy & pasting what `git --no-pager diff` from build log in CircleCI.

File: docs/source/R-api.rst [hunk 2]
```diff
@@ -17,7 +17,7 @@ You can use the R API to `install MLflow <install_mlflow_>`_, start the `user in
 
 Install MLflow
 
-Installs auxiliary dependencies of MLflow (e.g. the MLflow CLI). As a
+Installs auxiliary dependencies of MLflow (e.g. the MLflow CLI). As a
```
Comment (harupy) [inline docs/source/R-api.rst:20]: pandoc inserts a non-breaking whitespace after `e.g.` and `i.e.`.

File: .circleci/config.yml [hunk 3]
```diff
@@ -56,6 +56,7 @@ jobs:
           name: Show diff
           command: |
             git --no-pager diff
+            git diff --quiet
```
Comment (dbczumar) [inline .circleci/config.yml:59]: For context, why are both of these lines necessary?

File: mlflow/R/mlflow/document.R (hunk 1 above)
Comment (dbczumar) [reply mlflow/R/mlflow/document.R:52]: Fantastic! :D

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: .circleci/config.yml (hunk 3 above)
Comment (harupy) [reply .circleci/config.yml:59]: I added `git diff --quiet` to make this step fail when there are differences, but I just notice we can use `--exit-code
`.

```suggestion
            git --no-pager diff --exit-code
```

Comment (harupy) [reply .circleci/config.yml:59]: Fixed!

**Pull request merged**